cpp/include/cudf/search.hpp

```diff
@@ -0,0 +1,132 @@
+ *      idx        0    1    2    3    4
+ *   t      = {{  10,  20,  20,  20,  20 },
+ *             { 5.0,  .5,  .5,  .7,  .7 },
+ *             {  90,  77,  78,  61,  61 }}
+ *   values = {{ 20 },
+ *             { .7 },
+ *             { 61 }}
+ *   result =  {  3 }
+ * 
+ * @param t             Table to search
+ * @param values        Find insert locations for these values
+ * @param column_order  Vector of column sort order
+ * @param null_precedence The size of a NULL value in comparison to all other
+ * values
+ * @param mr Device memory resource to use for device memory allocation
+ * @return std::unique_ptr<column> A non-nullable column of INT32 elements
+ * containing the insertion points.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<column> lower_bound(table_view const& t,
+                                    table_view const& values,
+                                    std::vector<order> const& column_order,
+                                    null_order null_precedence,
+                                    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**---------------------------------------------------------------------------*
+ * @brief Find largest indices in a sorted table where values should be 
+ *  inserted to maintain order
+ * 
+ * For each row v in @p values, find the last index in @p t where
+ *  inserting the row will maintain the sort order of @p t
+ * 
+ * Example:
+ * 
+ *  Single Column:
+ *      idx      0   1   2   3   4
+ *   column = { 10, 20, 20, 30, 50 }
+ *   values = { 20 }
+ *   result = {  3 }
+ * 
+ *  Multi Column:
+ *    idx        0    1    2    3    4
+ *   t      = {{  10,  20,  20,  20,  20 },
+ *             { 5.0,  .5,  .5,  .7,  .7 },
+ *             {  90,  77,  78,  61,  61 }}
+ *   values = {{ 20 },
+ *             { .7 },
+ *             { 61 }}
+ *   result =  {  5  *   * 
+ * @param column        Table to search
+ * @param values        Find insert locations for these values
+ * @param column_order  Vector of column sort order
+ * @param null_precedence The size of a NULL value in comparison to all other
+ * values
+ * @param mr Device memory resource to use for device memory allocation
+ * @return std::unique_ptr<column> A non-nullable column of INT32 elements
+ * containing the insertion points.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<column> upper_bound(table_view const& t,
+                                    table_view const& values,
+                                    std::vector<order> const& column_order,
+                                    null_order null_precedence,
+                                    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**---------------------------------------------------------------------------*
+ * @brief Find if the `value` is present in the `column` and dtype of both
```

Comment:
No need to specify the type require in the brief. 
```suggestion
 * @brief Find if the `value` is present in `col`.
```